### PR TITLE
Testing for authentication#030

### PR DESCRIPTION
**Pull request by Salim-54**
#### What does this PR do?

- Tests authentication API endpoints

#### Description of Task to be completed? 

This is to test authentication API endpoints

#### How should this be manually tested?

- Clone the repo

- checkout to branch ``cr-authentication-api-tests#030``

- Run `npm install` to install modules

- Run `npm  run start` to start the development server

- In your API testing tool (postman recommended) call api `localhost:3000/api/v1/users` to add a todo task

#### Any background context you want to provide?

- N/A

#### What are the relevant Pivotal Tracker/trello stories?

- https://trello.com/c/dsiVJQQv/48-authentication-tests030

#### Screenshots (if appropriate)
- N/A

#### Questions:

- N/A

### CODE REVIEW

**Review by PrinceNiyonshuti — APPROVED** [review]
Looks Good

**Review by Xldivin — APPROVED** [review]

File: src/database/model/user.model.js [hunk 1]
```diff
@@ -28,18 +28,18 @@ const userSchema = new mongoose.Schema({
...
+  //   type: String,
+  //   required: [true, 'Please confirm your password'],
+  //   validate: {
+  //     // This only works on CREATE and SAVE!!!
+  //     validator: function(el) {
+  //       return el === this.password;
+  //     },
+  //     message: 'Passwords are not the same!'
+  //   }
+  // }
```
Comment (thepatrickniyo) [inline src/database/model/user.model.js:42]: Take care of comments

File: src/database/model/user.model.js [hunk 2]
```diff
@@ -50,8 +50,8 @@ userSchema.pre('save', async function(next) {
   this.password = await bcrypt.hash(this.password, 12);
 
   //delete pw
-  this.passwordConfirm = undefined;
-  next();
+  // this.passwordConfirm = undefined;
+  // next();
```
Comment (thepatrickniyo) [inline src/database/model/user.model.js:54]: take care of comments

**Review by thepatrickniyo — CHANGES_REQUESTED** [review]
Remove the comments

**Pull request merged**